src/disk_io_thread.cpp

```diff
@@ -1767,7 +1767,7 @@ namespace libtorrent
 		j->storage = storage->shared_from_this();
 		j->buffer.string = allocate_string_copy(p.c_str());
 		j->callback = std::move(handler);
-		j->flags = flags;
+		j->flags = std::uint8_t(flags);
```

Comment:
in all of these cases of ``flags``, I think it would be better to change the type of the argument. That way the information about the truncation is propagated up the call tree and allows warnings to be triggered earlier